src/main/java/org/folio/fqm/repository/DataRefreshRepository.java

```diff
@@ -0,0 +1,162 @@
+package org.folio.fqm.repository;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.fqm.client.SimpleHttpClient;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record2;
+import org.jooq.impl.DSL;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.table;
+
+@Repository
+@RequiredArgsConstructor
+@Log4j2
+public class DataRefreshRepository {
+
+  public static final Field<String> CURRENCY_FIELD = field("currency", String.class);
+  public static final Field<Double> EXCHANGE_RATE_FIELD = field("exchange_rate", Double.class);
+
+  private static final String REFRESH_MATERIALIZED_VIEW_SQL = "REFRESH MATERIALIZED VIEW CONCURRENTLY ";
+  private static final String GET_EXCHANGE_RATE_PATH = "finance/exchange-rate";
+  private static final String GET_LOCALE_SETTINGS_PATH = "configurations/entries";
+  private static final Map<String, String> GET_LOCALE_SETTINGS_PARAMS = Map.of(
+    "query", "(module==ORG and configName==localeSettings)"
+  );
+
+  private static final List<String> MATERIALIZED_VIEW_NAMES = List.of(
+    "drv_circulation_loan_status",
+    "drv_inventory_item_status",
+    "drv_pol_payment_status",
+    "drv_pol_receipt_status",
+    "drv_inventory_statistical_code_full",
+    "drv_languages"
+  );
+
+  static final List<String> SYSTEM_SUPPORTED_CURRENCIES = List.of(
+    "USD",
+    "JPY",
+    "BGN",
+    "CZK",
+    "DKK",
+    "GBP",
+    "HUF",
+    "PLN",
+    "RON",
+    "SEK",
+    "CHF",
+    "ISK",
+    "NOK",
+    "HRK",
+    "RUB",
+    "TRY",
+    "AUD",
+    "BRL",
+    "CAD",
+    "CNY",
+    "HKD",
+    "IDR",
+    "ILS",
+    "INR",
+    "KRW",
+    "MXN",
+    "MYR",
+    "NZD",
+    "PHP",
+    "SGD",
+    "THB",
+    "ZAR"
+  );
```

Comment:
These pretty much have to be hardcoded, unless we want to duplicate all the functionality in mod-finance.